community_tasks/ger_rag_eval.py

```diff
@@ -0,0 +1,224 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# ruff: noqa: F405, F403, F401
+"""
+Custom evaluation tasks for lighteval. Copy this file and complete it with the info for your task.
+
+This file generally create just a TASKS_TABLE and TASKS_GROUPS which are then imported by LightEval.
+
+Author:
+"""
+import numpy as np
+from aenum import extend_enum
+
+from lighteval.metrics import Metrics
+from lighteval.metrics.metrics import SampleLevelMetric
+from lighteval.metrics.utils import MetricCategory, MetricUseCase
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.tasks_prompt_formatting import LETTER_INDICES
+
+
+task1 = LightevalTaskConfig(
+    name="ger_rag_eval_task1",
+    prompt_function="prompt_fn_task1",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task1",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task2 = LightevalTaskConfig(
+    name="ger_rag_eval_task2",
+    prompt_function="prompt_fn_task2",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task2",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task3 = LightevalTaskConfig(
+    name="ger_rag_eval_task3",
+    prompt_function="prompt_fn_task3",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task3",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task4 = LightevalTaskConfig(
+    name="ger_rag_eval_task4",
+    prompt_function="prompt_fn_task4",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task4",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+QUERY_TASK1: str = """\
```

Comment:
Please set these prompts directly in the associated functions

Reply:
done